lib/new_relic/harvest/telemetry_sdk/dimensional_metrics/harvester.ex

```diff
@@ -0,0 +1,87 @@
+defmodule NewRelic.Harvest.TelemetrySdk.DimensionalMetrics.Harvester do
+  use GenServer
+
+  @moduledoc false
+
+  alias NewRelic.Harvest
+  alias NewRelic.Harvest.TelemetrySdk
+
+  @interval_ms 5_000
+
+  @valid_types [:count, :gauge, :summary]
+
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, [])
+  end
+
+  def init(_) do
+    {:ok,
+     %{
+       start_time_ms: System.system_time(:millisecond),
+       metrics: []
+     }}
+  end
+
+  # API
+
+  @spec report_dimensional_metric(:count | :gauge | :summary, atom() | binary(), any, map()) ::
```

Comment:
Should I remove the typespec?

Reply:
I feel like a pretty detailed spec would be worth it on the function inside `NewRelic` module, I'd go beyond what's here to help folks understand what the `value` should be for the various types